File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        const int size = nums.size();
        if (size == 0) { return false; }
        sort(nums.begin(), nums.end());
        int i = 0;

        for (i; i < size - 1; i++) {
            if (nums[i] == nums[i + 1]) { return true; }
        }

        return false;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-3.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        n = len(nums)
        st = set()
        count = {}

        for num in nums:
            count[num] = count.get(num, 0) + 1
            if count[num] > n // 3:
                st.add(num)

        return list(st)